applications/tests/test_applications_new_schema_ws_queries.py

```diff
@@ -0,0 +1,389 @@
+                        ),
+                        "customer": None,
+                    }
+                }
+            ]
+        }
+    }
+
+
+@pytest.mark.parametrize(
+    "api_client",
+    ["berth_services", "berth_handler", "berth_supervisor"],
+    indirect=True,
+)
+def test_winter_storage_applications_no_customer_filter_false(
+    winter_storage_application, api_client
+):
+    winter_storage_application.customer = None
+    winter_storage_application.save()
+
+    query = WS_APPLICATIONS_WITH_NO_CUSTOMER_FILTER_QUERY % "false"
+
+    executed = api_client.execute(query)
+
+    assert executed["data"] == {"winterStorageApplications": {"edges": []}}
+
+
+WINTER_STORAGE_APPLICATIONS_WITH_STATUSES_FILTER_QUERY = """
+query APPLICATIONS {
+    winterStorageApplications(statuses: [%s]) {
+        edges {
+            node {
+                id
+                status
+            }
+        }
+    }
+}
+"""
+
+
+@pytest.mark.parametrize(
+    "api_client",
+    ["berth_services", "berth_handler", "berth_supervisor"],
+    indirect=True,
+)
+def test_winter_storage_applications_statuses_filter(
+    handled_ws_application, api_client
+):
+
+    status_enum_str = ApplicationStatus.HANDLED.name
+
+    query = WINTER_STORAGE_APPLICATIONS_WITH_STATUSES_FILTER_QUERY % status_enum_str
+
+    executed = api_client.execute(query)
+
+    assert executed["data"] == {
+        "winterStorageApplications": {
+            "edges": [
+                {
+                    "node": {
+                        "id": to_global_id(
+                            WinterStorageApplicationNode._meta.name,
+                            handled_ws_application.id,
+                        ),
+                        "status": status_enum_str,
+                    }
+                }
+            ]
+        }
+    }
+
+
+@pytest.mark.parametrize(
+    "api_client",
+    ["berth_services", "berth_handler", "berth_supervisor"],
+    indirect=True,
+)
+def test_winter_storage_applications_statuses_filter_empty(
+    handled_ws_application, api_client
+):
+    status_enum_str = ApplicationStatus.PENDING.name
+
+    query = WINTER_STORAGE_APPLICATIONS_WITH_STATUSES_FILTER_QUERY % status_enum_str
+
+    executed = api_client.execute(query)
+
+    assert executed["data"] == {"winterStorageApplications": {"edges": []}}
+
+
+def test_winter_storage_applications_statuses_filter_invalid_enum(
+    handled_ws_application, superuser_api_client
+):
+    nonexistent_enum_str = "FOOBAR"
+
+    query = (
+        WINTER_STORAGE_APPLICATIONS_WITH_STATUSES_FILTER_QUERY % nonexistent_enum_str
+    )
+
+    executed = superuser_api_client.execute(query)
+
+    assert_in_errors(
+        "invalid value [%s]." % nonexistent_enum_str, executed,
+    )
+
+
+@pytest.mark.parametrize(
+    "api_client",
+    ["berth_services", "berth_handler", "berth_supervisor"],
+    indirect=True,
+)
+def test_winter_storage_applications_statuses_filter_empty_list(
```

Comment:
You can also use the `superuser_api_client` here. Since you have already tested that the query _works_, and this is just a variation of the query, it doesn't matter if you don't explicitly check for all the possible user groups.